### PR TITLE
Enable renovate

### PR DESCRIPTION
**Pull request by dashpole**
This is based on the renovate config used for otel-go and for collector-contrib.

### CODE REVIEW

**Comment by dashpole** [comment]
It seems hard to test this locally. I'm inclined to just merge it, and see what it generates.

File: renovate.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      "config:recommended"
+    ],
+    "ignorePaths": [],
+    "labels": ["Skip Changelog", "dependencies"],
+    "postUpdateOptions" : [
+      "gomodTidy"
+    ],
+    "postUpdateTask" : {
+        "commands": ["make", "fixtures"]
+    },
```
Comment (aabmass) [inline renovate.json:13]: FYI this won't work unless we use self-hosted renovate https://docs.renovatebot.com/configuration-options/#postupgradetasks:~:text=Post%2Dupgrade%20tasks%20can%20only%20be%20used%20on%20self%2Dhosted%20Renovate%20instances.

**Review by aabmass — APPROVED** [review]
Are we moving from dependabot to renovate then?

File: renovate.json (hunk 1 above)
Comment (dashpole) [reply renovate.json:13]: Got it.  Lets skip for now.

**Comment by dashpole** [comment]
I think we should align with whatever the upstream version of our repository does.  We can lean on work done upstream to keep this config working (e.g. when new versions break things).  Since OTel-go uses renovate, it seems good to do the same here.

File: renovate.json (hunk 1 above)
Comment (dashpole) [reply renovate.json]: removed

**Pull request merged**